src/borg/cache.py

```diff
@@ -924,19 +929,24 @@ def check_cache_compatibility(self):
         return True
 
     def wipe_cache(self):
-        logger.warning("Discarding incompatible cache and forcing a cache rebuild")
+        logger.warning("Discarding incompatible or corrupted cache and forcing a cache rebuild")
         archive_path = os.path.join(self.path, "chunks.archive.d")
         if os.path.isdir(archive_path):
             shutil.rmtree(os.path.join(self.path, "chunks.archive.d"))
             os.makedirs(os.path.join(self.path, "chunks.archive.d"))
         self.chunks = ChunkIndex()
-        with SaveFile(os.path.join(self.path, files_cache_name()), binary=True):
+        with IntegrityCheckedFile(path=os.path.join(self.path, "chunks"), write=True) as fd:
+            self.chunks.write(fd)
```

Comment:
What exactly is it saving to disk here? What's the contents of the chunks hashtable?

Reply:
The chunks hashtable saved to disk is a valid empty ChunkIndex here thanks to the previous line `self.chunks = ChunkIndex()`
This is similar to the LocalCache.create() function behavior. IntegrityCheckedFile doesn't interfere and just allows us to pass the integrity check when calling open() later.
Does it answer your question ?

Reply:
Can you check if it does a chunks cache resync?

If borg continues and does not exit there, it must do that before working with that (empty) chunks hashtable. If it exits, it needs to do the resync the next time borg is used.

Reply:
Just checked. The resync was not triggered because of CacheConfig.save() that overrides the manifest_id with the provided manifest.
I just pushed a commit where the config is written as-is with an empty manifest_id. This correctly triggers a call to sync() in  LocalCache.__init__()

Reply:
So it resyncs the chunks cache in same borg run before using the chunks cache?

Reply:
I fixed the failing test and pushed the commit. Then I slighlty modified the expected string to make it fail again, so I get to see the output:

E       AssertionError: assert 'xxxxx forcing a cache rebuild' in """
Discarding incompatible or corrupted cache and forcing a cache rebuild
Repository ID: caaff3f127e80db9d19a24762e4f21c7870c3bc7b6066bdb37f9c67771a80f3c
Location: /private/var/folders/61/qtqwzmxd7t1fb19zck5k3q0h0000gn/T/pytest-of-tw/pytest-88/test_cache_chunks0/repository
Repository version: 2
Append only: False
Encrypted: Yes (repokey AES-OCB)
Storage quota: 460 B used
Cache: /private/var/folders/61/qtqwzmxd7t1fb19zck5k3q0h0000gn/T/pytest-of-tw/pytest-88/test_cache_chunks0/cache/caaff3f127e80db9d19a24762e4f21c7870c3bc7b6066bdb37f9c67771a80f3c
Security dir: /private/var/folders/61/qtqwzmxd7t1fb19zck5k3q0h0000gn/T/pytest-of-tw/pytest-88/borg-base-dir0/.local/share/borg/security/caaff3f127e80db9d19a24762e4f21c7870c3bc7b6066bdb37f9c67771a80f3c
Original size: 0 B
Deduplicated size: 0 B
Unique chunks: 0
Total chunks: 0
"""

The last few lines seem to indicate that it **does not** do a chunks cache resync yet, but happily proceeds with the empty chunks cache.

Reply:
Yes on the same run.

Reply:
Let me check again

Reply:
It seems that the test do not create an archive so there is nothing in the cache. If we do a create_src_archive(archiver, "test") for example inside the test_cache_chunks() we can clearly see that the cache is not empty and correctly synced after corruption.

Reply:
That might be, but borg needs to work correctly for all commands.

Commands must not work with a "fake" chunks cache that is out of sync with repo reality.

Reply:
Yes I agree. On all my testing it works for any commands (rinfo, create, ...)

If I remove the "chunks" corruption in test_cache_chunks() I get this result : 
```
Repository ID: 610198ddd42c7a3ac2160ae0a965fdbff2b11a199440e717d24a3dd1c8d342b3
Location: /tmp/pytest-of-aztorius/pytest-13/test_cache_chunks0/repository
Repository version: 2
Append only: False
Encrypted: Yes (repokey AES-OCB)
Storage quota: 460 B used
Cache: /tmp/pytest-of-aztorius/pytest-13/test_cache_chunks0/cache/610198ddd42c7a3ac2160ae0a965fdbff2b11a199440e717d24a3dd1c8d342b3
Security dir: /tmp/pytest-of-aztorius/pytest-13/borg-base-dir1/.local/share/borg/security/610198ddd42c7a3ac2160ae0a965fdbff2b11a199440e717d24a3dd1c8d342b3
Original size: 0 B
Deduplicated size: 0 B
Unique chunks: 0
Total chunks: 0
```
So the cache is as expected. I do not see any issue.

Reply:
Ah, ok, so that test is not yet suitable to test for the cache sync (as one can't see the difference) - can you modify it so the repo isn't empty when the cache gets corrupted?

Reply:
I just pushed a commit to create an archive before corruption.
Maybe we should add an assert on the chunks file content to effectively check if everything is in sync ?

Reply:
Guess making a "copy" of the chunks hashtable before the corruption and then comparing all keys/value after the chunks cache resync has happened would be best.

We could also check if the resync happens (iirc it tells that on the output).

Reply:
New commit pushed with a check on chunks hashtable content.

Reply:
looks good.